runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoInfoBoxOverlay.java

```diff
@@ -62,48 +63,51 @@ private InfernoJadOverlay(final Client client, final InfernoPlugin plugin, final
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		if (!plugin.isShowPrayerHelp() || (plugin.getPrayerOverlayMode() != InfernoPrayerOverlayMode.BOTTOM_RIGHT
-			&& plugin.getPrayerOverlayMode() != InfernoPrayerOverlayMode.BOTH))
+		if (plugin.getPrayerDisplayMode() != InfernoPrayerDisplayMode.BOTTOM_RIGHT
+			&& plugin.getPrayerDisplayMode() != InfernoPrayerDisplayMode.BOTH)
 		{
 			return null;
 		}
 
-		InfernoJad.Attack attack = null;
-		int leastTicks = 999;
+		imagePanelComponent.getChildren().clear();
 
-		for (InfernoJad jad : plugin.getJads())
+		if (plugin.getClosestAttack() != null)
 		{
-			if (jad.getNextAttack() == null || jad.getTicksTillNextAttack() < 1)
-			{
-				continue;
-			}
+			final BufferedImage prayerImage = getPrayerImage(plugin.getClosestAttack());
 
-			if (jad.getTicksTillNextAttack() < leastTicks)
-			{
-				leastTicks = jad.getTicksTillNextAttack();
-				attack = jad.getNextAttack();
-			}
+			imagePanelComponent.getChildren().add(new ImageComponent(prayerImage));
+			imagePanelComponent.setBackgroundColor(client.isPrayerActive(plugin.getClosestAttack().getPrayer())
+				? ComponentConstants.STANDARD_BACKGROUND_COLOR
+				: NOT_ACTIVATED_BACKGROUND_COLOR);
 		}
-
-		if (attack == null)
+		else
 		{
-			return null;
+			imagePanelComponent.setBackgroundColor(ComponentConstants.STANDARD_BACKGROUND_COLOR);
 		}
 
-		final BufferedImage prayerImage = getPrayerImage(attack);
-
-		imagePanelComponent.getChildren().clear();
-		imagePanelComponent.getChildren().add(new ImageComponent(prayerImage));
-		imagePanelComponent.setBackgroundColor(client.isPrayerActive(attack.getPrayer())
-			? ComponentConstants.STANDARD_BACKGROUND_COLOR
-			: NOT_ACTIVATED_BACKGROUND_COLOR);
-
 		return imagePanelComponent.render(graphics);
 	}
 
-	private BufferedImage getPrayerImage(InfernoJad.Attack attack)
+	private BufferedImage getPrayerImage(InfernoNPC.Attack attack)
```

Comment:
Instead of getting a new image from spritemanager every time, they should probably be lazily initialized (like the hydra plugin does it, for example)
```
	BufferedImage getImage(SpriteManager spriteManager)
	{
		if (image == null)
		{
			image = spriteManager.getSprite(spriteID, 0);
		}

		return image;
	}